### ISSUE TITLE
Separate LUA layer

### ISSUE THREAD

**Issue by andrsd** [Open-Sn/opensn#9]
There is a LUA layer in the code that is used to drive the simulation. Currently this layer is compiled into the library (i.e. the computation heart of the code).

This layer needs to be refactored away from the computation and become its own library that is linked to the driver layer, i.e. binary that drives the computation.

This will improve internal design and will be very useful for scenarios where OpenSn will be used for in-memory coupling.  This will also enable to compile the library without LUA, i.e. when users just need the raw power of the library. Finally, this will also help in future if somebody will want to use another scripting language to drive their simulation (python, etc.)

**Comment by ragusa** [comment]
I agree. Note that with lua, we were able to connect a TCP port and listen in. This was useful for GUI's and later for digital twins where an action from a user or an update sensor variable can be read in during the simulation. (interaction mode you see at the very beginning of the main). This is a useful feature we should keep in mind and retain. This is only an FYI as I think we should be able to achieve this with lua, python, java, ...

**Comment by andrsd** [comment]
This work should not remove any capabilities you already have. Just move them in a different spot. Just to clarify...

**Comment by ragusa** [comment]
done
